src/core/qgsconditionalstyle.cpp

```diff
@@ -0,0 +1,104 @@
+#include <QPainter>
+
+#include "qgsconditionalstyle.h"
+#include "qgsexpression.h"
+#include "qgsfontutils.h"
+#include "qgssymbollayerv2utils.h"
+#include "qgsmarkersymbollayerv2.h"
+
+QgsConditionalStyle::QgsConditionalStyle()
+    : mValid( false )
+    , mSymbol( 0 )
+{}
+
+QgsConditionalStyle::QgsConditionalStyle( QString rule )
+    : mValid( false )
+    , mSymbol( 0 )
+{
+  setRule( rule );
+}
+
+void QgsConditionalStyle::setSymbol( QgsSymbolV2* value )
+{
+  mSymbol = value;

```

Comment:
this should probably take a clone of the symbol here, and delete the symbol in the destructor (or make mSymbol a QScopedPointer). Then you'll also need to add a copy and assignment constructor which also clones the symbol.